lib/fetch/index.js

```diff
@@ -30,6 +30,20 @@ const { ReadableStream } = require('stream/web')
 const { performance } = require('perf_hooks')
 const EE = require('events')
 
+// https://fetch.spec.whatwg.org/#garbage-collection
+let registry = []
+setInterval(function prune () {
+  const next = []
+  for (const entry of registry) {
+    if (entry.ref.deref() === undefined) {
+      entry.abort()
+    } else {
+      next.push(entry)
+    }
+  }
+  registry = next
+}, 1e3).unref()
```

Comment:
We could use a finalizationRegistry as well but that would be a lot less deterministic and the user might queue up lot's of requests before GC would start up.

Reply:
The number of elements in the "registry" is limited by the number of active connections so it should normally be quite small.

Reply:
@mcollina wdyt? deterministic polling loop or undeterministic `FinalizationRegistry`?

Reply:
The polling is unfeasible, it will keep processes active even with no actual traffic. Let's go with FinalizationRegistry.

Reply:
Fixed. PTAL.